File: api-common/src/main/java/com/ricky/apicommon/utils/result/R.java
package com.ricky.apicommon.utils.result;

import java.io.Serializable;
import java.util.Date;

public class R<T> implements Serializable {
    public T data;
    public  boolean success;
    public String msg;
    public Date datetime;

    public R(T data, boolean success, String msg) {
        this.data = data;
        this.success = success;
        this.msg = msg;
        datetime = new Date();
    }
}

File: file-centre/src/utils/R.ts
export class R {
  data: any;
  success: boolean;
  msg: string;
  datetime: Date;

  constructor(data: any, success: boolean, msg: string) {
    this.data = data;
    this.success = success;
    this.msg = msg;
    this.datetime = new Date();
  }

  static success(data: any = null, msg: string = 'ok'): R {
    return new R(data, true, msg);
  }

  static fail(msg: string, data: any = null): R {
    return new R(data, false, msg);
  }
}
